File: 基礎題庫/Contents/d471/d471.cpp
#include<iostream>

int num;
char bin[16];

void dfs(int);

using namespace std;

int main(void)
{
	while(cin >> num)
	{
		bin[num] = 0;
		dfs(0);
	}
	
	return 0;
}

void dfs(int n)
{
	if(n == num)
	{
		cout << bin << endl;
		return;
	}
	
	bin[n] = '0';
	dfs(n + 1);
	bin[n] = '1';
	dfs(n + 1);
}

File: 基礎題庫/Contents/d471/d471.py
import sys

def dfs(n):
    if n == num:
        print(*binary, sep="")
        return None
    
    binary[n] = '0'
    dfs(n + 1)
    binary[n] = '1'
    dfs(n + 1)

for num in sys.stdin:
    num = int(num)
    binary = [0] * num
    dfs(0)
